jdaviz/core/tools.py

```diff
@@ -93,12 +88,32 @@ def deactivate(self):
     def on_limits_change(self, *args):
         # from_lims: limits in the viewer belonging to the tool
         from_lims = {k: getattr(self.viewer.state, k) for k in self.match_keys}
+        orig_refdata = self.viewer.state.reference_data
 
         for viewer in self._iter_matched_viewers(include_self=False):
             # orig_lims: limits in this "matched" viewer
             # to_lims: proposed new limits for this "matched" viewer
             orig_lims = {k: getattr(viewer.state, k) for k in self.match_keys}
             to_lims = self._map_limits(self.viewer, viewer, from_lims)
+
+            to_refdata = viewer.state.reference_data
+
+            if orig_refdata != to_refdata and hasattr(self.viewer, '_get_fov'):
```

Comment:
this mixin is also used for spectrum-viewers... is this useful there too or do we need to check the viewer type?

Reply:
The `hasattr` check only passes for image viewers, so this shouldn't be a problem.